metrics_proto.go

```diff
@@ -91,20 +207,41 @@ func (se *statsExporter) handleMetricsProtoUpload(payloads []*metricProtoPayload
 
 	for _, payload := range payloads {
 		// Now create the metric descriptor remotely.
-		if err := se.createMetricDescriptor(ctx, payload.metric, payload.additionalLabels); err != nil {
-			span.SetStatus(trace.Status{Code: 2, Message: err.Error()})
-			return err
+		if payload.metric.GetMetricDescriptor().GetType() == metricspb.MetricDescriptor_SUMMARY {
+			payload.convertedMetrics = se.convertSummaryMetrics(payload.metric)
```

Comment:
Consider converting summary metrics at `ExportMetricsProto`: https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/pull/152/files#diff-c4657249ba7aeb2b6dc05cd7ec12497eR74, then you may be able to get rid of the conditional checks here and below.

Reply:
good point. Let me try.